### PR TITLE
VOOD-62: Avoid reloading of panel.

### PR DESCRIPTION
**Pull request by GinoCardillo-OST**

### CODE REVIEW

**Review by PascalSchuermannOst — COMMENTED** [review]
I get from my Linter the error:
Unsafe assignment of an `any` value. For line 44 and 94 in the file debugEventManager.ts
Can we resolve this Issue or disabe eslint for this lines?

**Comment by GinoCardillo-OST** [comment]
@PascalSchuermannOst The build wouldn't succeed if there was a Linter error. I don't have any errors
image
Must be something on your end.
My guess is you didn't run `npm install` to install the object-hash library and their types because the two lines you mentioned both use this.

**Comment by PascalSchuermannOst** [comment]
> 
> 
> @PascalSchuermannOst The build wouldn't succeed if there was a Linter error. I don't have any errors image Must be something on your end. Did you run `npm install` to install the object-hash library and their types? Because the two lines you mentioned both use this.

The ESLint plugin for my IDE had some issues, the errors disappeared after a few reconfigurations.
sorry for any inconvenience.

**Review by PascalSchuermannOst — APPROVED** [review]

File: src/debug-adapter/debugEventManager.ts [hunk 1]
```diff
@@ -1,25 +1,166 @@
...
+      panelViewVariable = { id: id !== 'null' ? id : hash(variable) };
+      if (!parentId) {
+        panelViewVariable.name = variable.name;
+      }
+
+      if (variable.type !== 'null') {
+        panelViewVariable.type = variable.type;
+        if (variable.type === 'String') {
+          [panelViewVariable.value, panelViewVariable.tooltip] = this.prepareStringData(variable);
+        } else if (this.primitiveDataArray.includes(variable.type)) {
```
Comment (alexandre-lagadec-ost) [inline src/debug-adapter/debugEventManager.ts:103]:
```suggestion
        } else if (this.primitiveArrayDataTypes.includes(variable.type)) {
```

File: src/debug-adapter/debugEventManager.ts [hunk 2]
```diff
@@ -1,25 +1,166 @@
 import { debug } from 'vscode';
+import { PanelViewInput, PanelViewVariable } from '../model/panelViewInput';
+import { Variable } from '../model/variable';
 import { DebuggerPanel } from '../webview/debuggerPanel';
 import { DebugSessionProxy } from './debugSessionProxy';
+import * as hash from 'object-hash';
 
 export class DebugEventManager {
+  private readonly primitiveDataArray = ['boolean[]', 'char[]', 'byte[]', 'short[]', 'int[]', 'long[]', 'float[]', 'double[]'];
```
Comment (alexandre-lagadec-ost) [inline src/debug-adapter/debugEventManager.ts:9]:
```suggestion
  private readonly primitiveArrayDataTypes = ['boolean[]', 'char[]', 'byte[]', 'short[]', 'int[]', 'long[]', 'float[]', 'double[]'];
```

**Review by alexandre-lagadec-ost — APPROVED** [review]

**Pull request merged**